WordPress/gutenberg
# Fix/55278 standardize font library rest api

## What?
Fixes #55278

* Refactors the Font Library REST URLs to use more standard locations
* Refactors the Font Library REST endpoints to use SINGULAR items instead of PLURAL items
* Uses `rest_ensure_response` for all API responses
* Included error handling of any JSON decoding
* adds schemas to the APIs
* 1 controller for Font Families, 1 for Font Collections
* Translate all error codes
* View changes to support SINGULAR responses and new error responses.

## Why?
The original implementation was "non standard" in a number of ways (documented in #55278)

## How?
Much of the work was previously accomplished during the [attempt to merge](https://github.com/WordPress/wordpress-develop/pull/5285) this work into WordPress Core for the 6.4 release.  Those changes have been ported here (as well as the view changes necessary).

## Testing Instructions
* Upload a Font via the upload UI.  It should be activated and useable on your site.
* Upload multiple fonts at a time.  They should all be available.
* Attempt to upload fonts from different families at the same time.  It should fail with an error message.
<img width="873" alt="image" src="https://github.com/WordPress/gutenberg/assets/146530/64730b2c-940c-4eac-9ff5-44c6b1713263">
Delete the uploaded font(s).  It should be removed from the system and no longer available.
<img width="869" alt="image" src="https://github.com/WordPress/gutenberg/assets/146530/467a68ab-6d08-420f-bf05-94fade47d9aa">
Install a font from Google Fonts. It should be activated and useable on your site.
<img width="872" alt="image" src="https://github.com/WordPress/gutenberg/assets/146530/e431f908-6ed1-4cef-89b6-4b24d775859e">
